main.go

```diff
@@ -150,20 +150,32 @@ func handlePullRequest(w http.ResponseWriter, body []byte, git Git, githubClient
 		message := fmt.Sprintf("Getting commits for PR %s/%s#%d failed", pullRequestEvent.Repository.Owner, pullRequestEvent.Repository.Name, pullRequestEvent.IssueNumber)
 		return ErrorResponse{err, http.StatusBadGateway, message}
 	}
+	if includeFixupCommits(commits) {
+		pr, _, err := githubClient.PullRequests.Get(pullRequestEvent.Repository.Owner, pullRequestEvent.Repository.Name, pullRequestEvent.IssueNumber)
+		if err != nil {
+			message := fmt.Sprintf("Getting PR %s/%s#%d failed", pullRequestEvent.Repository.Owner, pullRequestEvent.Repository.Name, pullRequestEvent.IssueNumber)
+			return ErrorResponse{err, http.StatusBadGateway, message}
+		}
+		_, _, err = githubClient.Repositories.CreateStatus(pullRequestEvent.Repository.Owner, pullRequestEvent.Repository.Name, *pr.Head.SHA, &github.RepoStatus{
+			State:       github.String("pending"),
+			Description: github.String("This PR contains commits that need to be squashed with !squash before merging"),
+			Context:     github.String(githubStatusContext),
+		})
+		if err != nil {
+			message := fmt.Sprintf("Failed to create a pending status for commit %s", *pr.Head.SHA)
+			return ErrorResponse{err, http.StatusBadGateway, message}
+		}
+	}
+	return SuccessResponse{}
+}
+
+func includeFixupCommits(commits []github.RepositoryCommit) bool {
 	for _, commit := range commits {
 		if strings.HasPrefix(*commit.Commit.Message, "fixup! ") || strings.HasPrefix(*commit.Commit.Message, "squash! ") {
```

Comment:
I don't think git supports commits starting with, say `FIXUP! `. I don't think we should either